src/main/java/seedu/address/logic/commands/FindCommand.java

```diff
@@ -1,43 +1,95 @@
+            + "Example: " + COMMAND_WORD + " John Cena";
+
+    public static final String MESSAGE_TASK_ASSIGNED = "%1$s has been assigned to the following tasks:";
+    public static final String MESSAGE_NO_TASK_ASSIGNED = "%1$s does not have any assigned tasks.";
 
-    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Finds all persons whose names contain any of "
-            + "the specified keywords (case-insensitive) and displays them as a list with index numbers.\n"
-            + "Parameters: KEYWORD [MORE_KEYWORDS]...\n"
-            + "Example: " + COMMAND_WORD + " alice bob charlie";
+    private final NameContainsExactKeywordsPredicate predicate;
 
-    private final NameContainsKeywordsPredicate predicate;
+    /**
+     * Creates FindCommand object with given personIndex
+     */
+    public FindCommand(NameContainsExactKeywordsPredicate predicate) {
+        requireAllNonNull(predicate);
 
-    public FindCommand(NameContainsKeywordsPredicate predicate) {
         this.predicate = predicate;
     }
 
+    /**
+     * Executes FindCommand
+     */
     @Override
-    public CommandResult execute(Model model, OfficeConnectModel officeConnectModel) {
-        requireNonNull(model);
-        model.updateFilteredPersonList(predicate);
-        return new CommandResult(
-                String.format(Messages.MESSAGE_PERSONS_LISTED_OVERVIEW, model.getFilteredPersonList().size()));
+    public CommandResult execute(Model model, OfficeConnectModel officeConnectModel) throws CommandException {
+        requireAllNonNull(model, officeConnectModel);
+
+        List<Person> personList = model.getAddressBook()
+                .getPersonList()
+                .filtered(predicate);
+
+        if (personList.size() != 1) {
+            throw new CommandException(Messages.MESSAGE_INVALID_PERSON);
+        }
+
+        Person person = personList.get(0);
+        Id pId = person.getId();
+        Name name = person.getName();
+
+        ObservableList<AssignTask> assignedTaskList = getAssignedTaskList(officeConnectModel, pId);
+
+        displayAssignedTaskAndPerson(model, officeConnectModel, assignedTaskList, pId);
+
+        if (assignedTaskList.isEmpty()) {
+            return new CommandResult(String.format(MESSAGE_NO_TASK_ASSIGNED, name));
+        } else {
+            return new CommandResult(String.format(MESSAGE_TASK_ASSIGNED, name));
+        }
+    }
+
+    private static void displayAssignedTaskAndPerson(Model model, OfficeConnectModel officeConnectModel,
+                                                     ObservableList<AssignTask> assignedTaskList, Id pId) {
+        RepositoryModelManager<Task> taskModelManager = officeConnectModel.getTaskModelManager();
+        model.updateFilteredPersonList(person -> person.getId().equals(pId));
+        taskModelManager.updateFilteredItemList(task -> assignedTaskList.stream()
+                .anyMatch(personTask -> personTask.getTaskId().equals(task.getId())));
```

Comment:
would it be better to use the officeConnectModel.updateTaskModelManagerFilteredItemList() here, rather than using taskModelManager?